internal/captain/values.go

```diff
@@ -122,35 +131,78 @@ type PackageValue struct {
 
 var _ FlagMarshaler = &PackageValue{}
 
-func (u *PackageValue) String() string {
-	if u.Namespace == "" && u.Name == "" {
+func (p *PackageValue) String() string {
+	if p.Namespace == "" && p.Name == "" {
 		return ""
 	}
-	if u.Version == "" {
-		return fmt.Sprintf("%s/%s", u.Namespace, u.Name)
+	name := p.Name
+	if p.Namespace != "" {
+		name = fmt.Sprintf("%s/%s", p.Namespace, p.Name)
 	}
-	return fmt.Sprintf("%s/%s@%s", u.Namespace, u.Name, u.Version)
+	if p.Version == "" {
+		return name
+	}
+	return fmt.Sprintf("%s@%s", name, p.Version)
 }
 
-func (u *PackageValue) Set(s string) error {
+func (p *PackageValue) Set(s string) error {
 	if strings.Contains(s, "@") {
 		v := strings.Split(s, "@")
-		u.Version = strings.TrimSpace(v[1])
+		p.Version = strings.TrimSpace(v[1])
 		s = v[0]
 	}
 	if strings.Index(s, "/") == -1 {
-		return fmt.Errorf("invalid package name format: %s (expected '<namespace>/<name>[@version]')", s)
+		p.Name = strings.TrimSpace(s)
+		return nil
 	}
 	v := strings.Split(s, "/")
-	u.Namespace = strings.TrimSpace(strings.Join(v[0:len(v)-1], "/"))
-	u.Name = strings.TrimSpace(v[len(v)-1])
+	p.Namespace = strings.TrimSpace(strings.Join(v[0:len(v)-1], "/"))
+	p.Name = strings.TrimSpace(v[len(v)-1])
+	return nil
+}
+
+func (p *PackageValue) Type() string {
+	return "package"
+}
+
+// PackageValueNoVersion is identical to PackageValue except that it does not support a version.
```

Comment:
For these "is identical to PackageValue" types, maybe it's possible to have some config fields in PackageValue instead:
```
type Package struct {
    Namespace string
    Name      string
    Version   string

    IsVersionForbidden  bool 
    IsNamespaceRequired bool
}

And then have the Set and Type logic behavior based on the values.

Reply:
I tried this but it just makes things more difficult as it would require us to manually initialize these structs and configure them before we can use them. That's possible, but it feels better to centralize the boilerplate in this file rather than throughout our cmdtree.